### DISCUSSION TITLE
Add CLI options for SSL configuration to directus init.

### DISCUSSION THREAD

**Discussion by cory-glooh** [directus/directus#16232] [Feature Requests]
As it stands right now, setting up directus with a remote managed database that uses SSL with a certificate., You can't use directus init, you need to either build the .env yourself, or select sqlite and then edit it afterwards. 

Would be so much nicer if you could toggle DB_SSL__REJECT_UNAUTHORIZED and provide a file path to the certificate, to auto-set the DB_SSL__CA fields.

**Comment by azrikahar** [comment]
> Would be so much nicer if you could toggle DB_SSL__REJECT_UNAUTHORIZED and provide a file path to the certificate, to auto-set the DB_SSL__CA fields.

It would be nice indeed, perhaps can be a follow up enhancement to #16093.

In regards to having a file path to the certificate, currently we can only set the string directly [such as mentioned here](https://github.com/directus/directus/discussions/7155#discussioncomment-1128448), but maybe one potential alternative if you'd like to use file path in some way, would be to use [config.js](https://docs.directus.io/self-hosted/config-options.html#config-js) so that you can do something such as:

```js
const fs = require('fs');

module.exports = {
    // ...
    DB_SSL__CA: fs.readFileSync("the-ca-file.pem"),
    // etc
};
```
> you need to either build the .env yourself

Out of curiosity, what is the current exact step(s) that you're doing for this?

Slightly off topic, but as it seems like currently createEnv (which will create the `.env` file based on the liquid template) is called in the `init` command:

https://github.com/directus/directus/blob/2adbb3bfd27639b3af4453d4ffcdd3be59736dea/api/src/cli/commands/init/index.ts#L69

I wonder would it make the [Setup a Configuration File](https://docs.directus.io/self-hosted/installation/manual.html#_3-setup-a-configuration-file) step easier to have another command that can be used to purely create said `.env` file directly and work from there, rather than having to select sqlite first. Of course this is not meant as the fix for cases such as `DB_SSL__REJECT_UNAUTHORIZED` prompt, but as a more generic alternative.

**Reply by cory-glooh** [reply]
> In regards to having a file path to the certificate, currently we can only set the string directly [such as mentioned here](https://github.com/directus/directus/discussions/7155#discussioncomment-1128448), but maybe one potential alternative if you'd like to use file path in some way, would be to use [config.js](https://docs.directus.io/self-hosted/config-options.html#config-js) so that you can do something such as:
> 
> ```js
> const fs = require('fs');
> 
> module.exports = {
>     // ...
>     DB_SSL__CA: fs.readFileSync("the-ca-file.pem"),
>     // etc
> };
> ```

My thought behind the path, is the init could then read that file and set the env apropriately. so we don't have to worry about formatting it properly. 

> > you need to either build the .env yourself
> 
> Out of curiosity, what is the current exact step(s) that you're doing for this?

At the moment I just select SQLite and then edit the .env and run bootstrap. 

> Slightly off topic, but as it seems like currently createEnv (which will create the `.env` file based on the liquid template) is called in the `init` command:
> 
> https://github.com/directus/directus/blob/2adbb3bfd27639b3af4453d4ffcdd3be59736dea/api/src/cli/commands/init/index.ts#L69
> 
> I wonder would it make the [Setup a Configuration File](https://docs.directus.io/self-hosted/installation/manual.html#_3-setup-a-configuration-file) step easier to have another command that can be used to purely create said `.env` file directly and work from there, rather than having to select sqlite first. Of course this is not meant as the fix for cases such as `DB_SSL__REJECT_UNAUTHORIZED` prompt, but as a more generic alternative.

Most definately.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
